### PR TITLE
onionshare: 2.2 -> 2.3.1

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
  Upstream release. [Changelog](https://github.com/micahflee/onionshare/releases/tag/v2.3)

The structure of the nix expression changed quite a bit, but that's because the structure of upstream did too, and for the better. They introduced Flatpak and Snap as their main method of distribution. Since those are declarative, this makes our job easier.

###### Things done

- [x] Wait for final release

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Is this read to be undraftet?

**Comment by lourkeur** [comment]
> Is this read to be undraftet?

I plan to undraft it when the final release drops.

**Comment by SuperSandro2000** [comment]
@lourkeur ping

**Comment by lourkeur** [comment]
> @lourkeur ping

2.3 is still not out. Thank you for your interest.

File: pkgs/applications/networking/onionshare/default.nix [hunk 1]
```diff
@@ -7,24 +7,28 @@
...
   pysocks,
   pytest,
+  qrcode,
   qt5,
   requests,
   tor,
   obfs4,
 }:
 
 let
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:25]: Please remove this let in. It no longer servers any purpose.

File: pkgs/applications/networking/onionshare/default.nix [hunk 2]
```diff
@@ -51,63 +55,70 @@ let
...
       requests
     ];
+
     buildInputs = [
       tor
       obfs4
     ];
 
+    checkInputs = [
+      pytest
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:89]: Please use pytestCheckHook.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 103844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>onionshare</li>
<li>onionshare-gui</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

onionshare-gui:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/applications/networking/onionshare/default.nix (hunk 1 above)
Comment (lourkeur) [reply pkgs/applications/networking/onionshare/default.nix:26]: It still serves a purpose: both derivations use `src`

**Comment by lourkeur** [comment]
https://github.com/micahflee/onionshare/releases/tag/v2.3.1

File: pkgs/applications/networking/onionshare/fix-paths.patch [hunk 3]
```diff
@@ -72,63 +35,6 @@ index 3373462..7fd245b 100644
 +        tor_geo_ip_file_path = "@geoip@/share/tor/geoip"
 +        tor_geo_ipv6_file_path = "@geoip@/share/tor/geoip6"
 +        obfs4proxy_file_path = "@obfs4@/bin/obfs4proxy"
-
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/fix-paths.patch:38]:
```suggestion

```

File: pkgs/applications/networking/onionshare/default.nix [hunk 4]
```diff
@@ -51,63 +56,77 @@ let
     maintainers = with maintainers; [ lourkeur ];
   };
 
-  common = buildPythonApplication {
-    pname = "onionshare-common";
-    inherit version meta src;
+
+in rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:60]:
```suggestion
in rec {
```

File: pkgs/applications/networking/onionshare/default.nix [hunk 5]
```diff
@@ -51,63 +56,77 @@ let
...
+    inherit version meta;
+    src = "${src}/cli";
+    patches = [
+      (substituteAll {
+        src = ./fix-paths.patch;
+        inherit tor obfs4;
+        inherit (tor) geoip;
+      })
+    ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:73]:
```suggestion

```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 103844` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>onionshare-gui</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>onionshare: <a href="https://termbin.com/mr2co">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/93cba66a068ea4b693df51baa2ef0183/raw/index.html">pretty log </a></li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
onionshare:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/onionshare/default.nix:65:7:

```
   |
65 |       (substituteAll {
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/onionshare/default.nix:103:7:

```
    |
103 |       (substituteAll {
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `stdenv`.
Near pkgs/applications/networking/onionshare/default.nix:4:3:

```
  |
4 |   stdenv,
  |   ^
```
Unused argument: `pysocks`.
Near pkgs/applications/networking/onionshare/default.nix:16:3:

```
   |
16 |   pysocks,
   |   ^
```
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 103844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>onionshare: <a href="https://termbin.com/yykb">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/2df7d1474655a813700582b737e5075b/raw/index.html">pretty log </a></li>
<li>onionshare-gui: log was empty</li>
  </ul>
</details>

File: pkgs/applications/networking/onionshare/fix-paths.patch (hunk 3 above)
Comment (lourkeur) [reply pkgs/applications/networking/onionshare/fix-paths.patch]: That broke the build, patch format is like that

**Comment by lourkeur** [comment]
@SuperSandro2000 fixed the hammering lints

**Pull request merged**